### PR TITLE
Add implementation of API product creation

### PR DESCRIPTION
**Pull request by dewmini**

### CODE REVIEW

**Comment by sajinidesilva** [comment]
Its better if you can squash these two commits in to one as it doesn't make any sense to keep the second commit as a separate one.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/product/add.jag [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        if (log.isDebugEnabled()) {
+            log.debug("creating API product");
+        }
+
+        return {
+            error: false
+        };
+    } catch (e) {
+        log.error(e);
```
Comment (sajinidesilva) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/product/add.jag]: Shouldn't we log the error here with customized error message?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/blocks/product-create/ajax/add.jag [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                var apiProductIdentifier = new APIProductIdentifier(provider, productName, productVersion);
+                var APIProduct = Packages.org.wso2.carbon.apimgt.api.model.APIProduct;
+                var apiProduct = new APIProduct(apiProductIdentifier);
+
+                apiProduct.setDescription(description);
+                apiProduct.setCreatedUser(provider);
+                apiProduct.setUpdatedUser(provider);
+                apiProduct.setBusinessOwner(businessOwner);
+                apiProduct.setBusinessOwnerEmail(businessOwnerEmail);
+                //apiProduct.setVisibility("");
```
Comment (sajinidesilva) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/blocks/product-create/ajax/add.jag]: Can we remove these commented lines?

Comment (dewmini) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/blocks/product-create/ajax/add.jag]: Fixed.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/product/add.jag (hunk 1 above)
Comment (dewmini) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/product/add.jag]: Fixed in the code.

**Comment by harsha89** [comment]
I believe this pull request is invalid now. Hence I'm closing it. If it required, please send a pull request again.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/item-manage/template.jag [hunk 3]
```diff
@@ -261,7 +263,55 @@
...
+                                <div class="col-sm-8">
+                                    <label class="radio" for="bundleProduct">
+                                       <input type="radio" class="product_options" id ="bundleProduct" name="product_options"
+                                       checked="checked" value="bundleProduct">
+                                       <span class="helper"><%=i18n.localize("Bundle with an existing API Product")%></span>
+                                    </label><br>
+                                    <div class="form-group productList"><br>
+                                        <label class="control-label col-xs-3"><%=i18n.localize("API Products")%>:
+                                        </label>
+                                        <div class="col-lg-5">
```
Comment (dewmini) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/item-manage/template.jag]: Fix is in https://github.com/dewmini/carbon-apimgt/commit/31e5bed7b95384e6753051bd58a25cf3d1abbd01

**Pull request closed without merging**